repose-aggregator/components/services/atom-feed-service/atom-feed-service-impl/src/main/scala/org/openrepose/nodeservice/atomfeed/impl/actors/FeedReader.scala

```diff
@@ -128,20 +128,22 @@ class FeedReader(feedURIString: String,
           authenticatedRequestFactory,
           authenticationTimeout)
 
-        val entryStream = order match {
+        val startingStream = getStream
```

Comment:
Considering this, I'm wondering why `getStream` is not just a variable. We always call it exactly once, so it's not like we are extracting shared logic or like we need it to be lazy.

I would even consider using an actual `var` for the stream since after manipulating it, we don't really care about tracking the previous values anymore. Also, it's about as local of state as you can get, so I'm not really worried about it escaping.